content/integrations/postgresql.md

```diff
@@ -47,58 +47,79 @@ When it prompts for a password, enter the one used in the first command.
-* `tags` (Optional) - Dictionary of tags
-* `relations` (Optional) - Dictionary of per-relation (table) metrics that you want to track. Each relation generates 10 metrics + 10 metrics per index.
-  
-  By default all schemas are included. To track relations from specific schemas only, use the following syntax:
+* `username` (Optional) - The user account used to collect metrics, set in the [Installation section above](#installation)
+* `password` (Optional) - The password for the user account.
+* `dbname` (Optional) - The name of the database you want to monitor.
+* `ssl` (Optional) - Defaults to False. Indicates whether to use an SSL connection.
+* `use_psycopg2` (Optional) - Defaults to False. Setting this option to `True` will force the Datadog Agent to collect PostgreSQL metrics using psycopg2 instead of pg8000. Note that pyscopg2 does not support SSL connections.
+* `tags` (Optional) - A list of tags applied to all metrics collected. Tags may be simple strings or key-value pairs.
+* `relations` (Optional) - By default, all schemas are included. Add specific schemas here to collect metrics for schema relations. Each relation will generate 10 metrics and an additional 10 metrics per index. Use the following structure to declare relations:
 
       relations:
-        - relation_name: another_table
+        - relation_name: my_relation
           schemas:
-            - public
-            - prod
+            - my_schema_1
+            - my_schema_2
   {:.language-yaml}
+
 * `collect_function_metrics` (Optional) - Collect metrics regarding PL/pgSQL functions from pg_stat_user_functions
-* `collect_count_metrics` (Optional) - Collect count metrics, default value is True for backward compatibility but they migth be slow, suggested value is False.
+* `collect_count_metrics` (Optional) - Collect count metrics. The default value is `True` for backward compatibility, but this might be slow. The recommended value is `False`.
 
 
 ## Custom metrics
- 
-Below are some examples of commonly used metrics, which are implemented as custom metrics. Uncomment them if you want to use them as is, or use as an example for creating your own custom metrics. 
 
-The format for describing custome metrics is identical with the one used for common metrics in `postgres.py`. Be extra careful with ensuring proper custom metrics description format. If your custom metric does not work after an agent restart, look for errors in the output of `/etc/init.d/datadog-agent info` command, as well as `/var/log/datadog/collector.log` file.
+PostgreSQL custom metrics can be collected by mapping SQL query results to metrics in Datadog. This mapping is expressed using four components: descriptors, metrics, query, and relation.
+
+* **Descriptors** are a sequence of tuples matching a column name or alias from the custom metric query to a tag in Datadog. In the example below, `relname` is mapped to `schema`. Custom metrics in Datadog can then be filtered and aggregated by `schema` tag.
```

Comment:
s/sequence/list

Descriptors **is**? Since this refers to an option name. Otherwise, "are" does agree with "a sequence".

Again, the 'matching' and 'mapping' words are throwing me. 'matching to a tag in Datadog': what tag in Datadog? What we are actually doing is creating a tag and naming it what we like.

How about:

**Descriptors** let you tag the custom metric with any of the query's column names or aliases. Each descriptor is a pair: the first element is the column name or alias, and the second element is the tag name. In the example below, the metric will be tagged: `schema:<relname>`.